### PR TITLE
feat: new system table: system.catalogs

### LINKED ISSUE #8418: Feature: add a new system table: system.catalogs
**Summary**

This allows users to query catalogs in the Databend system.

related: #8255

### PR DESCRIPTION
**Pull request by ClSlaid**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR implements a new system table `system.catalogs`.

图片

Related #8255 

Fixes #8418

### CODE REVIEW

**Review by PsiACE — APPROVED** [review]

File: src/query/catalog/src/table_context.rs [hunk 1]
```diff
@@ -71,6 +71,7 @@ pub trait TableContext: Send + Sync {
     fn try_set_partitions(&self, partitions: Partitions) -> Result<()>;
     fn attach_query_str(&self, kind: String, query: &str);
     fn get_fragment_id(&self) -> usize;
+    fn list_catalogs(&self) -> Vec<String>;
```
Comment (Xuanwo) [inline src/query/catalog/src/table_context.rs:74]: It's weird to have a `list_catalogs` API in `TableContext`.

File: src/query/storages/preludes/src/system/catalogs_table.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#[async_trait::async_trait]
+impl AsyncSystemTable for CatalogsTable {
+    const NAME: &'static str = "system.catalogs";
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn get_full_data(&self, ctx: Arc<dyn TableContext>) -> Result<DataBlock> {
+        let catalogs = ctx.list_catalogs();
```
Comment (Xuanwo) [inline src/query/storages/preludes/src/system/catalogs_table.rs:47]: `list_catalogs` from TableContext seems not cool. Maybe we should fetch it from the global catalog manager directly? 

Cc @sundy-li

File: src/query/service/tests/it/storages/system.rs [hunk 3]
```diff
@@ -203,6 +205,14 @@ async fn test_credits_table() -> Result<()> {
     Ok(())
 }
 
+async fn test_catalogs_table(file: &mut impl Write) -> Result<()> {
```
Comment (Xuanwo) [inline src/query/service/tests/it/storages/system.rs:208]: Please also add a stateless test.

File: src/query/storages/preludes/src/system/catalogs_table.rs (hunk 2 above)
Comment (sundy-li) [reply src/query/storages/preludes/src/system/catalogs_table.rs:47]: Yes.  The table context needs to be simply for table API usage.

File: src/query/service/tests/it/storages/system.rs (hunk 3 above)
Comment (sundy-li) [reply src/query/service/tests/it/storages/system.rs:208]: we do not need stateless test for system tables. It's hard to use.  It's easy to cover in golden file tests.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**